packages/autorest.python/autorest/codegen/serializers/builder_serializer.py

```diff
@@ -1069,12 +1069,18 @@ def response_headers_and_deserialization(
                 deserialize_code.append("    pipeline_response")
                 deserialize_code.append(")")
             elif self.code_model.options["models_mode"] == "dpg":
+                rest_filed = (
+                    f', rf=_RestField(format="{response.type.encode}")'
+                    if response.need_rest_field
+                    and isinstance(response.type, ByteArraySchema)
```

Comment:
multiple checks on response type being `ByteArraySchema` since `need_rest_field` already checks this